src/api/index.ts

```diff
@@ -16,16 +16,33 @@ import {
   erc20Allowances,
   FEE_TOKEN,
   exchangeOrders,
+  unregisteredTokens,
+  TOKEN_8,
 } from '../../test/data'
 import Web3 from 'web3'
 import { INITIAL_INFURA_ENDPOINT } from 'const'
 
+const isWeb3Mock = process.env.MOCK_WEB3 === 'true'
 const isWalletMock = process.env.MOCK_WALLET === 'true'
 const isTokenListMock = process.env.MOCK_TOKEN_LIST === 'true'
 const isErc20Mock = process.env.MOCK_ERC20 === 'true'
 const isDepositMock = process.env.MOCK_DEPOSIT === 'true'
 const isExchangeMock = process.env.MOCK_EXCHANGE === 'true'
 
+// TODO connect to mainnet if we need AUTOCONNECT at all
+export const getDefaultProvider = (): string | null =>
+  process.env.NODE_ENV === 'test' ? null : INITIAL_INFURA_ENDPOINT
+
+function createWeb3Api(): Web3 {
+  const web3 = new Web3(getDefaultProvider())
+
+  if (isWeb3Mock) {
```

Comment:
You didn't really need the whole `createWeb3Api` function. Can just move `if (isWeb3Mock) {` logic to L98

Reply:
I'd like to keep it here and refactor when doing https://github.com/gnosis/dex-react/issues/331.

I'll add a TODO in the code with the issue link as well.